src/main/java/org/opensearch/knn/index/mapper/KNNVectorFieldMapper.java

```diff
@@ -65,6 +67,15 @@ public abstract class KNNVectorFieldMapper extends ParametrizedFieldMapper {
     public static final String CONTENT_TYPE = "knn_vector";
     public static final String KNN_FIELD = "knn_field";
 
+    private static final Map<KNNEngine, KNNFlag> ENGINE_TO_STATS_SUPPLIER = ImmutableMap.of(
```

Comment:
Can we move the values present in this map to KNNEngine enum so that we don't need to maintain this mapping and we can get the KNNFlag from the KNNEngine enum itself.

Reply:
it's good idea, let me do the change